packages/foundry/contracts/utils/ContangoWand.sol

```diff
@@ -0,0 +1,208 @@
+import "../interfaces/ICauldron.sol";
+import "../interfaces/ILadleGov.sol";
+import "../interfaces/ILadle.sol";
+import "../interfaces/IJoin.sol";
+import "../oracles/yieldspace/YieldSpaceMultiOracle.sol";
+import "../oracles/composite/CompositeMultiOracle.sol";
+import "@yield-protocol/yieldspace-tv/src/interfaces/IPool.sol";
+import "@yield-protocol/utils-v2/src/access/AccessControl.sol";
+
+/// @title A contract that allows configuring the cauldron and ladle within bounds
+contract ContangoWand is AccessControl {
+    ICauldronGov public immutable cauldron;
+    ICauldron public immutable masterCauldron;
+    ILadleGov public immutable ladle;
+    ILadle public immutable masterLadle;
+    YieldSpaceMultiOracle public immutable yieldSpaceOracle;
+    CompositeMultiOracle public immutable compositeOracle;
+
+    mapping (bytes6 => mapping(bytes6 => uint32)) public ratio;
+    mapping (bytes6 => mapping(bytes6 => DataTypes.Debt)) public debt;
+
+    DataTypes.Debt public defaultDebtLimits;
+    uint32 public defaultRatio;
+
+    constructor(
+        ICauldronGov cauldron_,
+        ICauldron masterCauldron_,
+        ILadleGov ladle_,
+        ILadle masterLadle_,
+        YieldSpaceMultiOracle yieldSpaceOracle_,
+        CompositeMultiOracle compositeOracle_
+    ) {
+        cauldron = cauldron_;
+        masterCauldron = masterCauldron_;
+        ladle = ladle_;
+        masterLadle = masterLadle_;
+        yieldSpaceOracle = yieldSpaceOracle_;
+        compositeOracle = compositeOracle_;
+    }
+
+    /// ----------------- Cauldron Governance -----------------
+
+    /// @notice Copy the spotOracle and ratio from the master cauldron
+    function copySpotOracle(bytes6 baseId, bytes6 ilkId) external auth {
+        DataTypes.SpotOracle memory spotOracle_ = masterCauldron.spotOracles(baseId, ilkId);
+        cauldron.setSpotOracle(baseId, ilkId, spotOracle_.oracle, spotOracle_.ratio);
+    }
+
+    /// @notice Copy the lending oracle from the master cauldron
+    function copyLendingOracle(bytes6 baseId) external auth {
+        IOracle lendingOracle_ = masterCauldron.lendingOracles(baseId);
+        cauldron.setLendingOracle(baseId, lendingOracle_);
+    }
+
+    /// @notice Copy the debt limits from the master cauldron
+    function copyDebtLimits(bytes6 baseId, bytes6 ilkId) external auth {
+        DataTypes.Debt memory debt_ = masterCauldron.debt(baseId, ilkId);
+        cauldron.setDebtLimits(baseId, ilkId, debt_.max, debt_.min, debt_.dec);
+    }
+
+    /// @notice Add a new asset in the Cauldron, as long as it is an asset or fyToken known to the Yield Cauldron
+    function addAsset(bytes6 assetId) external auth {
+        address asset_ = masterCauldron.assets(assetId);
+        require(
+            asset_ != address(0) ||
+            address(masterCauldron.series(assetId).fyToken) != address(0),
+        "Asset not known to the Yield Cauldron");
+        cauldron.addAsset(assetId, asset_);
```

Comment:
This won't work for a fyToken, it'd add `address(0)` as they're not assets on the master cauldron